docs/non_native_field_arithmetic.md

```diff
@@ -0,0 +1,164 @@
+limbs of $r$ in a similar fashion:
+
+$$
+\begin{array}{l l}
+{{y_{0}=t_{0}+0,}}&{{y_{0}-r_{0}=z_{0}\cdot2^{B}}}\\ 
+{{y_{1}=t_{1}+z_{0},}}&{{y_{1}-r_{1}=z_{1}\cdot2^{B}}}\\ 
+{{y_{2}=t_{2}+z_{1},}}&{{y_{2}-r_{2}=z_{2}\cdot2^{B}}}\\ 
+{{y_{2}=t_{3}+z_{2},}}&{{y_{3}-r_{3}=z_{3}\cdot2^{B}}}
+\end{array}
+$$
+
+Removing the intermediate variables $y_i$, we get:
+
+$$
+\begin{array}{l}
+{{t_{0}+0-r_{0}=z_{0}\cdot2^{B}}}\\ 
+{{t_{1}+z_{0}-r_{1}=z_{1}\cdot2^{B}}}\\ 
+{{t_{2}+z_{1}-r_{2}=z_{2}\cdot2^{B}}}\\
+{{t_{3}+z_{2}-r_{3}=z_{3}\cdot2^{B}}}
+\end{array}
+$$
+
+To summarize, for a given $(\lbrack a_3,a_2,a_1,a_0\rbrack, \lbrack b_3, b_2, b_1, b_0\rbrack, \lbrack p'_3, p'_2, p'_1, p'_0\rbrack,
+\lbrack q_3, q_2, q_1, q_0\rbrack, \lbrack r_3,r_2,r_1,r_0\rbrack,
+t_3,t_2,t_1,t_0, z_3,z_2,z_1,z_0)$ that satisfies the 4 constraints above, along with constraints that ensures $p'=2^T-p$ (note that
+$p$ is an odd prime),
+
+$$
+\begin{array}{l}
+{{p_{0}^{\prime}+p_{0}+0=2^{B}}}\\ 
+{{p_{1}^{\prime}+p_{1}+1=2^{B}}}\\ 
+{{p_{2}^{\prime}+p_{2}+1=2^{B}}}\\ 
+{{p_{3}^{\prime}+p_{3}+1=2^{B}}}
+\end{array}
+$$
+
+it is guaranteed that $a \cdot b - q \cdot p - r = 0 \mod 2^T$ (but not $a \cdot b - q \cdot p - r = 0$ on their own).
+
+### Constraints in Our Native Field
+
+Next, we will consider a method to ensure $a \cdot b - q \cdot p - r = 0 \mod n$. We denote some new variables $\lbrace a_n, b_n, q_n,
+p_n,
+r_n\rbrace \ s.t. a_n = a \mod n$ etc. respectively. Now, we have more constraints with advice variables $\lbrace v_a, v_b, v_q, v_p,
+v_r, v_
+{overall}\rbrace$
+
+$$
+\begin{array}
+{c}{{v_{a}\cdot n+a_{n}=a}}\\ 
+{{v_{b}\cdot n+b_{n}=a}}\\ 
+{{v_{q}\cdot n+q_{n}=p}}\\ 
+{{v_{r}\cdot n+r_{n}=r}} \\
+v_{overall} \cdot n = a_n \cdot b_n - q_n \cdot p_n - r_n
+\end{array}
+$$
+
+Because $a,b,q,p,r$ are not fit in our native field, we need to decompose them into their limbs. 
+For a given $(a,b,q,p,r,v_a, v_b, v_q, v_p, v_r, v_{overall})$ that follow the 5 constraints above, it is guaranteed that $a \cdot b -
+q \cdot p - r = 0 \mod n$, but not $a \cdot b - q \cdot p - r = 0$.
+
+### Applying the Chinese Remainder Theorem
+
+Now we have $a \cdot b - q \cdot p - r = 0 \mod n$ and $a \cdot b - q \cdot p - r = 0 \mod 2^T$, which implies that $a \cdot b - q
+\cdot p - r = 0 \mod (n \cdot 2^T)$ by our CRT. If we choose $T$ such that $a \cdot b - q \cdot p - r < p^2 < n \cdot 2^T$, then it
+means that $a \cdot b - q \cdot p - r = 0$. Note that we should choose $T$ such that $t_3 < n$ to be able to represent all our numbers
+in our native field $\mathbb{F}_n$.
+
+## Non-native Field Arithmetic with Lookup Argument
+
+> [!Warning]
+> This section is written according to my personal understanding, so it should be used for reference rather than belief
+
+Because of the "Cost of non-aligned field operations have gone down a lot since the introduction of schemes like Plookup." sentence 
```

Comment:
because..... but ?
Did you check your grammar before pushing ?
Rewrite this paragraph.
WLS!